src/libraries/System.Numerics.Tensors/ref/System.Numerics.Tensors.netcore.cs

```diff
@@ -4,8 +4,237 @@
+    }
+}
 namespace System.Numerics.Tensors
 {
+    public partial interface ITensor<TSelf, T> : System.Collections.Generic.IEnumerable<T>, System.Collections.IEnumerable, System.IEquatable<TSelf>, System.Numerics.IEqualityOperators<TSelf, TSelf, bool> where TSelf : System.Numerics.Tensors.ITensor<TSelf, T>
+    {
+        static TSelf? Empty { get { throw null; } }
+        bool IsEmpty { get; }
+        bool IsPinned { get; }
+        ref T this[scoped System.ReadOnlySpan<nint> indices] { get; }
+        int Rank { get; }
+        System.ReadOnlySpan<nint> Strides { get; }
+        System.Numerics.Tensors.ReadOnlyTensorSpan<T> AsReadOnlyTensorSpan(scoped System.ReadOnlySpan<System.Buffers.NRange> ranges);
+        System.Numerics.Tensors.TensorSpan<T> AsTensorSpan(scoped System.ReadOnlySpan<System.Buffers.NRange> ranges);
+        void Clear();
+        void CopyTo(System.Numerics.Tensors.TensorSpan<T> destination);
+        void Fill(T value);
+        ref T GetPinnableReference();
+        static abstract bool operator ==(TSelf left, TSelf right);
+        static abstract bool operator ==(TSelf left, T right);
+        static abstract implicit operator System.Numerics.Tensors.ReadOnlyTensorSpan<T> (TSelf value);
+        static abstract implicit operator System.Numerics.Tensors.TensorSpan<T> (TSelf value);
+        static abstract bool operator !=(TSelf left, TSelf right);
+        static abstract bool operator !=(TSelf left, T right);
+        TSelf Slice(scoped System.ReadOnlySpan<System.Buffers.NRange> ranges);
+        bool TryCopyTo(System.Numerics.Tensors.TensorSpan<T> destination);
+    }
+    public readonly ref partial struct ReadOnlyTensorSpan<T>
+    {
+        private readonly object _dummy;
+        private readonly int _dummyPrimitive;
+        [System.CLSCompliantAttribute(false)]
+        public unsafe ReadOnlyTensorSpan(void* pointer, System.ReadOnlySpan<nint> lengths, System.ReadOnlySpan<nint> strides = default(System.ReadOnlySpan<nint>)) { throw null; }
+        public ReadOnlyTensorSpan(T[]? array, nint start, System.ReadOnlySpan<nint> lengths) { throw null; }
+        public ReadOnlyTensorSpan(T[]? array, System.ReadOnlySpan<nint> lengths) { throw null; }
+        public static System.Numerics.Tensors.ReadOnlyTensorSpan<T> Empty { get { throw null; } }
+        public bool IsEmpty { get { throw null; } }
+        public bool IsPinned { get { throw null; } }
+        public ref readonly T this[scoped System.ReadOnlySpan<nint> indices] { get { throw null; } }
+        public nint LinearLength { get { throw null; } }
+        public int Rank { get { throw null; } }
+        public System.ReadOnlySpan<nint> Shape { get { throw null; } }
+        public System.ReadOnlySpan<nint> Strides { get { throw null; } }
+        public static System.Numerics.Tensors.ReadOnlyTensorSpan<T> CastUp<TDerived>(System.Numerics.Tensors.ReadOnlyTensorSpan<TDerived> items) where TDerived : class?, T? { throw null; }
```

Comment:
I forgot to include this one in API review myself. I'll add it to the list to bring in the next review.